crates/sui-benchmark/src/options.rs

```diff
@@ -124,25 +125,56 @@ pub enum RunSpec {
     // and make progress in parallel. But this too
     // will likely change in future to support
     // more representative workloads.
+    //
+    // The Bench command allow us to define multiple benchmark groups in order to simulate different
+    // traffic characteristics across the whole benchmark duration. For that reason all arguments are
+    // expressed as vectors. Each benchmark group runs for the specified duration - as defined on the
+    // duration field - and for each group the parameters of the same vector position are considered.
+    // For example, for benchmark group 0, the vector arguments on position 0 refer to the properties
+    // of that benchmark group. The benchmark groups will run in a rotation fashion, unless the duration
+    // of the last group is set as "unbounded" which will run of the rest of the whole benchmark.
+    //
+    // Example: for Bench argument:
```

Comment:
I think defining a series of benchmarks via a configuration file would make this cleaner to setup and make it a little less error prone when running it via cli. Up to you if you want to change it now or later, but I think it be nice to define the benchmarks in a yaml and pass that in i.e.
```
benchmarks:
- target-qps: 1000
  shared_counter: 1
  duration: 360s
  ...
- target-qps: 2000
  transfer_object: 1
  duration: 120s
  ...
```

Reply:
Let me do that as a follow up PR as this will require more work and potentially more work on the CI flows as well assuming we want to structure the yaml files there differently too.